gateway/src/http_entrypoints.rs

```diff
@@ -298,6 +298,45 @@ struct UpdatePreparationProgress {
     total: u32,
 }
 
+/// Result of reading an SP sensor.
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    JsonSchema,
+)]
+pub struct SpSensorReading {
+    /// SP-centric timestamp.
```

Comment:
Maybe a bit more detail? Does it have units? Is it only useful as to compute deltas, or is the absolute value meaningful?

Reply:
I was intentionally vague here because it depends on what the SP is using as a timestamp mechanism, but maybe that's not very useful. Today, this is "milliseconds since the last SP boot", which I think is primarily useful to compute deltas (within the same SP as long as it hasn't rebooted), although the absolute value is incidentally meaningful (i.e., time since last boot). I don't know that this is a committed, stable guarantee on the SP side, though; if we started injecting NTP time into the SP, maybe this value would change, for example.

I'd be happy to expand the comment to explain what this value means today, maybe with a note that it could change with future SP releases?

Reply:
I think a quick note would be useful, maybe something about units and that it should probably only be used for deltas at this point. Out of curiosity, does the SP cache any measurements? Or is it reading a sensor directly in response to the request from the host?

Reply:
It is caching measurements; under the hood, this request sends a [SensorRequestKind::LastReading](https://github.com/oxidecomputer/management-gateway-service/blob/fda8faaff9774dafed08c995e19a402efdad0cfb/gateway-sp-comms/src/single_sp.rs#L807), which ultimately lands in the hubris sensor task here: https://github.com/oxidecomputer/hubris/blob/8c42218796329eb48c9d50c93b5db8aba776310a/task/sensor/src/main.rs#L104-L112. The way the `sensors` task is structured, I believe it cannot cause a sensor to be read; it can only report readings (as we're doing here) or accept posts of updated values from other tasks.

All that to say - if you request a value from the same sensor multiple times faster than whatever task is checking that sensor, you can definitely expect to get back identical results (including identical timestamps, since it's the timestamp of the last data posted to the `sensor` task).

Reply:
:+1: I put a longer comment in explaining what it is today, plus a note that it might change in a future SP release. Kinda vague, but maybe will serve as a hint if we change this in the future and it surprises someone.